ESPWebThingAdapter.h

```diff
@@ -362,16 +352,15 @@ class WebThingAdapter {
         request->beginResponseStream("application/json");
 
     DynamicJsonDocument buf(LARGE_JSON_DOCUMENT_SIZE);
-    JsonArray things = buf.to<JsonArray>();
+    JsonObject thing = buf.to<JsonObject>();
     ThingDevice *device = this->firstDevice;
     while (device != nullptr) {
-      JsonObject descr = things.createNestedObject();
-      device->serialize(descr, ip, port);
-      descr["href"] = "/things/" + device->id;
+      device->serialize(thing, ip, port);
+      thing["href"] = "/things/" + device->id;
       device = device->next;
     }
 
-    serializeJson(things, *response);
```

Comment:
Does this remove support for having multiple web things per device? I would actually support that change since I never liked that feature, but it's a bit strange to only support one thing but still use `things/` in the URL. I'd suggest that if you do this (and others agree), that the base URL for a thing's API (and the URL of its Thing Description) should be `/` rather than `/things/{id}`.

If the others do not agree and this feature is kept, we should in future consider supporting the [Directory Service API](https://w3c.github.io/wot-discovery/#directory-api-spec) and [`_directory._sub._wot`](https://w3c.github.io/wot-discovery/#introduction-dns-sd) DNS service name proposed in the WoT Discovery specification when providing a list of things.

Reply:
Yes, I removed that, because it produces an array as output, which isn't WoT compliant.
So, I didn't even really had another option than to remove it.
Well, yes, the whole naming schema in generally is not very ideal. This was an effort supported by Siemens.
The priority was to make it WoT compliant and not so much to clean everything up. There are a lot of things which can get improved. The whole: We put everything in two header files is also not very ideal. It's also not a very good coding convention. But as I said: Priority on migrating it.